File: adgeistkit/src/main/java/com/adgeistkit/ads/AdRequest.java
package com.adgeistkit.ads;

import org.json.JSONObject;

public class AdRequest {
    private final boolean testMode;

    private AdRequest(Builder builder) {
        this.testMode = builder.testMode;
    }

    public static class Builder {
        private boolean testMode;

        public Builder setTestMode(boolean testMode) {
            this.testMode = testMode;
            return this;
        }

        public AdRequest build() {
            return new AdRequest(this);
        }
    }

    public boolean isTestMode() {
        return testMode;
    }

    @Override
    public String toString() {
        return "AdRequest(testMode=" + testMode + ")";
    }

    public JSONObject toJson() {
        JSONObject json = new JSONObject();
        try { json.put("testMode", testMode); }
        catch (Exception ignored) {}
        return json;
    }
}

File: adgeistkit/src/main/java/com/adgeistkit/ads/network/AdRequest.kt
package com.adgeistkit.ads.network

import org.json.JSONObject

class AdRequest private constructor(builder: Builder) {
    val isTestMode: Boolean = builder.testMode

    class Builder {
        internal var testMode: Boolean = false

        fun setTestMode(testMode: Boolean): Builder {
            this.testMode = testMode
            return this
        }

        fun build(): AdRequest {
            return AdRequest(this)
        }
    }

    override fun toString(): String {
        return "AdRequest(testMode=" + isTestMode + ")"
    }

    fun toJson(): JSONObject {
        val json = JSONObject()
        try {
            json.put("testMode", isTestMode)
        } catch (ignored: Exception) {
        }
        return json
    }
}